wain-exec/src/cast.rs

```diff
@@ -22,103 +73,275 @@ macro_rules! float_to_int {
     };
 }
 
-float_to_int!(f32_to_u32, f32, u32);
-float_to_int!(f32_to_u64, f32, u64);
-float_to_int!(f64_to_u32, f64, u32);
-float_to_int!(f64_to_u64, f64, u64);
-float_to_int!(f32_to_i32, f32, i32);
-float_to_int!(f32_to_i64, f32, i64);
-float_to_int!(f64_to_i32, f64, i32);
-float_to_int!(f64_to_i64, f64, i64);
+float_to_int_sat!(f32_to_u32_sat, f32, u32);
+float_to_int_sat!(f32_to_u64_sat, f32, u64);
+float_to_int_sat!(f64_to_u32_sat, f64, u32);
+float_to_int_sat!(f64_to_u64_sat, f64, u64);
+float_to_int_sat!(f32_to_i32_sat, f32, i32);
+float_to_int_sat!(f32_to_i64_sat, f32, i64);
+float_to_int_sat!(f64_to_i32_sat, f64, i32);
+float_to_int_sat!(f64_to_i64_sat, f64, i64);
 
 #[cfg(test)]
 mod tests {
     use super::*;
     use std::{f32, f64};
 
+    trait NextAfter {
+        fn nextafter(self, _: Self) -> Self;
```

Comment:
単語の間は `_` で区切っておいていただけると．

```suggestion
        fn next_after(self, _: Self) -> Self;
```